design/deploy-steps.md

```diff
@@ -0,0 +1,150 @@
+<!--
+ This work is licensed under a Creative Commons Attribution 3.0
+ Unported License.
+
+ http://creativecommons.org/licenses/by/3.0/legalcode
+-->
+
+# Customizable deployment procedure
+
+## Status
+
+implementable
+
+## Summary
+
+This design proposes customizing the deployment procedure by providing a high
+level abstraction around Ironic [deploy steps][deploy-steps], as well as
+allowing per-host deployment (IPA) images.
+
+[deploy-steps]: https://docs.openstack.org/ironic/latest/admin/node-deployment.html#deploy-steps
+
+## Motivation
+
+By default the Ironic agent downloads a provided image via HTTP(s), converts it
+(if needed) and writes directly to the disk. This procedure works for many
+distributions, but may not suit some cases well. A prominent example is CoreOS,
+which has its own installer ([coreos-installer][coreos-installer]).
+
+[coreos-installer]: https://github.com/coreos/coreos-installer
+
+### Goals
+
+- A way to replace the default deploy procedure with a custom action.
+
+### Non-Goals
+
+- Expose the full functionality of Ironic deploy steps.
+- Add a way to execute custom actions during deployment.
+- A way to provide a link to the Ironic agent image per host.
+
+## Proposal
+
+### User Story
+
+As a CoreOS user I want to use `coreos-installer` to write the CoreOS image
+without losing any functionality that Ironic offers (unlike the live ISO
+workflow).
+
+## Design Details
+
+Additions to `BareMetalHost`:
+
+- New structure field `CustomDeploy` with the only field `Method` to engage
```

Comment:
bike-shed

```suggestion
- New structure field `Deploy` with the only field `Method` to engage
```

Reply:
I like that the `Custom` prefix highlights that this is an advanced feature, rather than something that people normally set.